Generate new API docs, add custom domain (#843)
rancher/rio: docs/continuous-deployment.md

You can do this with Git Basic Auth or SSH Auth:
- Git Basic Auth:
    1. Configure git basic auth credential secrets:
    
        ```bash
        $ rio secret create --git-basic-auth
        Select namespace[default]: $(put the same namespace with your workload)
        git url[]: $(for example: https://github.com)
        username[]: $(your GH username)
        password[******]: $(your GH password)
        ```
       
    2. Create a workload pointing to your repo using standard git checkout. For example:
        
        ```bash
        rio run -p 8080 https://github.com/example/example-private-repo
       ```
       
- SSH Auth:
    1. Configure git sshkey auth credential secrets. This should use a key that does not have a password associated to it:
       
        ```bash
        $ rio secret create --git-sshkey-auth
        Select namespace[default]: $(put the same namespace with your workload)
        git url[]: $(put your github url. Leave out http/https. Example: github.com)
        ssh_key_path[]: $(type the path to your ssh private key)
        ```
       

>     2. Create workload pointing to your repo using ssh checkout. For example:

Reviewer: ```suggestion
    2. Tell rio to do ssh checkout from git. For example:
```

        `rio run --build-clone-secret gitcredential-ssh -p 8080 git@github.com:example/example.git`

#### Private Docker Registry
1. Configure the Docker credential secret.

```bash
$ rio secret create --docker
Select namespace[default]: $(put the same namespace with your workload)
Registry URL[https://index.docker.io/v1/]: $(found with "docker info | grep Registry")
username[]: $(your docker username)
password[******]: $(your docker password)
```

> 2. Push your image to docker registry:

Reviewer: Can we also clarify that this also runs the built image?

Author: @rancher-max Feel like runs the built image shouldn't be here. It will be in the section of `running workload` with just imagePullSecrets

Reviewer: @StrongMonkey I'm fine with that. In that case should we remove the similar "create a workload" parts elsewhere in this file?

Author: Where is that? can you put suggestions in here? I saw other examples are just purely running http repo.

Reviewer: Okay I understand now. Made a few suggestions that might help other section clarity as well.
